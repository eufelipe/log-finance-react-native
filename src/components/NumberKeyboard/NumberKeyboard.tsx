import {Button, Currency, Strap} from 'components';
import React, {useState} from 'react';
import {useTranslation} from 'react-i18next';

import {
  Container,
  DismissArea,
  Grid,
  Key,
  EmptyKey,
  KeyLabel,
  ButtonContainer,
  BackspaceIcon,
  Touchable,
  Label,
  ContainnerValue,
  Value,
} from './styles';

interface Props {
  valueDefault?: number;
  onDismiss: () => void;
  onDone: (value: number) => void;
}

const MAX_NUMBER_SIZE = 8;

export const NumberKeyboard = ({
  onDone,
  onDismiss,
  valueDefault = 0,
}: Props): JSX.Element => {
  const [value, setValue] = useState(`${valueDefault}`);
  const {t} = useTranslation('global');

  const onBackSpace = () => {
    if (!value) return;
    setValue(value.toString().slice(0, -1));
  };

  const appendValue = (valueEntered: string) => {
    if (valueEntered === '0' && value === '0') return;
    if (valueEntered === '.' && value.indexOf('.') > -1) return;

    const sumValue = `${value}${valueEntered}`;

    if (sumValue.length === MAX_NUMBER_SIZE) return;

    setValue(sumValue);
  };

  return (
    <>
      <DismissArea onPress={onDismiss} />

      <Container>
        <ContainnerValue>
          <Currency
            value={Number(value)}
            render={text => <Value>{text}</Value>}
          />
          <Touchable onPress={() => setValue('')}>
            <Label>{t('clear')}</Label>
          </Touchable>
        </ContainnerValue>

        <Strap />

        <Grid>
          <Key onPress={() => appendValue('1')}>
            <KeyLabel>1</KeyLabel>
          </Key>
          <Key onPress={() => appendValue('2')}>
            <KeyLabel>2</KeyLabel>
          </Key>
          <Key onPress={() => appendValue('3')}>
            <KeyLabel>3</KeyLabel>
          </Key>
        </Grid>
        <Grid>
          <Key onPress={() => appendValue('4')}>
            <KeyLabel>4</KeyLabel>
          </Key>
          <Key onPress={() => appendValue('5')}>
            <KeyLabel>5</KeyLabel>
          </Key>
          <Key onPress={() => appendValue('6')}>
            <KeyLabel>6</KeyLabel>
          </Key>
        </Grid>

        <Grid>
          <Key onPress={() => appendValue('7')}>
            <KeyLabel>7</KeyLabel>
          </Key>
          <Key onPress={() => appendValue('8')}>
            <KeyLabel>8</KeyLabel>
          </Key>
          <Key onPress={() => appendValue('9')}>
            <KeyLabel>9</KeyLabel>
          </Key>
        </Grid>
        <Grid>
          <Key onPress={() => appendValue('.')}>
            <KeyLabel>,</KeyLabel>
          </Key>
          <Key onPress={() => appendValue('0')}>
            <KeyLabel>0</KeyLabel>
          </Key>
          <EmptyKey>
            <Touchable onPress={() => onBackSpace()}>
              <BackspaceIcon />
            </Touchable>
          </EmptyKey>
        </Grid>
        <ButtonContainer>
          <Button inverse title={t('cancel')} onPress={onDismiss} />
          <Button title={t('done')} onPress={() => onDone(Number(value))} />
        </ButtonContainer>
      </Container>
    </>
  );
};

export default NumberKeyboard;
