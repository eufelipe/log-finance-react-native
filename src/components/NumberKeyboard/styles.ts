import styled, {css} from 'styled-components/native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import {getBottomSpace} from 'react-native-iphone-x-helper';
import {TouchableOpacity, Text, Colors} from 'styles/layout';
import shadow from 'styles/shadow';
import {width} from 'styles/mixins';

const KEY_SIZE = 60;

const absolute = css`
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
`;

const centered = css`
  justify-content: center;
  align-items: center;
`;

const ContainerKey = css`
  width: ${KEY_SIZE}px;
  height: ${KEY_SIZE}px;
  border-radius: ${KEY_SIZE / 2}px;
`;

export const Container = styled.View`
  padding: 5px 0px ${({theme}) => theme.spacing.venti}px;
  background: ${({theme}) => theme.colors.accent};
  border-top-right-radius: ${({theme}) => theme.spacing.great}px;
  border-top-left-radius: ${({theme}) => theme.spacing.great}px;
  ${centered}
  ${absolute}
  ${shadow}
`;

export const ButtonContainer = styled.View`
  flex-direction: row;
  margin-top: ${({theme}) => theme.spacing.venti}px;
  margin-bottom: ${getBottomSpace()}px;
  ${centered}
`;

export const DismissArea = styled.TouchableOpacity.attrs({
  activeOpacity: 0.8,
})`
  flex: 1;
  top: 0;
  blur-radius: 90px;
  ${absolute}
`;

export const Grid = styled.View`
  flex: 1;
  flex-wrap: wrap;
  flex-direction: row;
  justify-content: flex-start;
`;

export const Key = styled(TouchableOpacity)`
  background: ${({theme}) => theme.colors.primary};
  margin: ${({theme}) => theme.spacing.small}px;
  ${centered}
  ${ContainerKey}
`;
export const EmptyKey = styled.View`
  background: transparent;
  margin: ${({theme}) => theme.spacing.small}px;
  ${centered}
  ${ContainerKey}
`;

export const KeyLabel = styled(Text)`
  color: ${({theme}) => theme.colors.white};
`;

export const Value = styled(Text).attrs({
  lines: 1,
})`
  color: ${({theme}) => theme.colors.white};
  font-size: ${({theme}) => theme.fontSizes.big};
`;

export const ContainnerValue = styled.View`
  flex-direction: row;
  justify-content: space-between;
  flex: 1
  width: ${width - 40}px;
  padding: ${({theme}) => theme.spacing.small}px 0px;
  border-bottom: 1px solid red;
`;

export const Touchable = styled(TouchableOpacity)`
  justify-content: center;
`;

export const Label = styled(Text)`
  color: ${({theme}) => theme.colors.white};
  font-size: ${({theme}) => theme.fontSizes.small};
`;

export const BackspaceIcon = styled(MaterialIcons).attrs({
  name: 'backspace',
  size: 24,
  color: Colors.white,
})`
  margin-left: ${({theme}) => theme.spacing.great}px;
`;
