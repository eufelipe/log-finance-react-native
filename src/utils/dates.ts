import {
  format as formatPattern,
  isValid,
  parseISO as parseISOFNS,
} from 'date-fns';

import locale from 'date-fns/locale/pt-BR';

export const DATE_FORMAT_ISO = 'yyyy-MM-dd';
export const DATE_FORMAT_BR = 'dd/MM/yyyy';
export const DATE_FORMAT_FRIENDLY = " dd/MM 'às' HH:mm";
export const DATE_FORMAT_FRIENDLY_TIME = "'às' HH:mm";

export function parseStringToDate(value: string): Date {
  const parsedDate = parseISOFNS(value);
  if (!isValid(parsedDate)) {
    throw Error('canot parse date');
  }
  return parsedDate;
}
export function parseDateToString({
  value,
  format = DATE_FORMAT_BR,
}: {
  value: Date;
  format?: string;
}): string {
  return formatPattern(value, format, {locale});
}

export const getDate = (date: Date): string => {
  return formatPattern(date, DATE_FORMAT_ISO, {locale});
};

export const getDateToday = (): string => {
  const today = new Date();
  return formatPattern(today, DATE_FORMAT_ISO, {locale});
};

export const getDateAndHourFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY, {locale});
};

export const getTimeFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY_TIME, {locale});
};
