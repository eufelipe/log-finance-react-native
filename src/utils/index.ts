export {default as number} from './number';
export {default as strings} from './strings';
