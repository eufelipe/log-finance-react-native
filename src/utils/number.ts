export const sanitizeToFloat = (value: string): number => {
  value = value.replace(/[^0-9.]+/, '');
  if (value.includes('.') && value.includes(',')) {
    value = value.replace(/\./g, '').replace(',', '.');
  } else if (value.includes(',')) {
    value = value.replace(',', '.');
  }
  return parseFloat(value) || 0;
};
