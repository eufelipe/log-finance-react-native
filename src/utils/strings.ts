export const sanitizeString = (value: string): string => {
  return value
    .toLowerCase()
    .replace(/á/g, 'a')
    .replace(/é/g, 'e')
    .replace(/í/g, 'i')
    .replace(/ó/g, 'o')
    .replace(/ú/g, 'u');
};
