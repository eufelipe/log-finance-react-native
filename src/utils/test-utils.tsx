/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {render, RenderAPI, RenderOptions} from '@testing-library/react-native';

import {theme} from 'styles/theme';
import {ThemeProvider} from 'styled-components/native';

import {NavigationContainer} from '@react-navigation/native';

interface Props {
  children: JSX.Element;
}

const AllTheProviders = ({children}: Props) => {
  return (
    <ThemeProvider theme={theme}>
      <NavigationContainer>{children}</NavigationContainer>
    </ThemeProvider>
  );
};

const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions,
): RenderAPI => render(ui, {wrapper: AllTheProviders, ...options});

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
