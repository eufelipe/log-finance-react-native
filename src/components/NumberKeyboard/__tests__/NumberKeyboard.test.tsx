/* eslint-disable @typescript-eslint/unbound-method */
/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {render} from 'utils/test-utils';
import NumberKeyboard from '../NumberKeyboard';

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon');

jest.mock('react-i18next', () => ({
  useTranslation: () => ({t: (key: any) => key}),
}));

const MOCK = {
  valueDefault: 0,
  onDismiss: jest.fn(),
  onDone: jest.fn(),
};

describe('NumberKeyboard', () => {
  it('Test match snapshot NumberKeyboard', () => {
    // given
    const props = MOCK;

    // when
    const {toJSON} = render(<NumberKeyboard {...props} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });
});
