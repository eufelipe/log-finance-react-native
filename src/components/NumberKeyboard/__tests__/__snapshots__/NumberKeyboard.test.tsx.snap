// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberKeyboard Test match snapshot NumberKeyboard 1`] = `
Array [
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "blurRadius": 90,
        "bottom": 0,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "left": 0,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#181D26",
          "borderTopLeftRadius": 21.428571428571427,
          "borderTopRightRadius": 21.428571428571427,
          "bottom": 0,
          "elevation": 10,
          "justifyContent": "center",
          "left": 0,
          "paddingBottom": 64.28571428571428,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 5,
          "position": "absolute",
          "right": 0,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 1,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottom": "1px solid red",
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "space-between",
            "paddingBottom": 10.714285714285714,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 10.714285714285714,
            "width": 710,
          },
        ]
      }
    >
      <Text
        lines={1}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Poppins-Regular",
              "fontSize": 60,
              "fontWeight": "400",
            },
          ]
        }
      >
        R$ 0,00
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Poppins-Regular",
                "fontSize": 28,
                "fontWeight": "400",
              },
            ]
          }
        >
          clear
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#464E5F10",
            "borderRadius": 2.5,
            "height": 5,
            "marginBottom": 10,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 10,
            "width": 100,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "flexWrap": "wrap",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2B303A",
            "borderRadius": 30,
            "height": 60,
            "justifyContent": "center",
            "marginBottom": 10.714285714285714,
            "marginLeft": 10.714285714285714,
            "marginRight": 10.714285714285714,
            "marginTop": 10.714285714285714,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Poppins-Regular",
                "fontSize": 36,
                "fontWeight": "400",
              },
            ]
          }
        >
          1
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2B303A",
            "borderRadius": 30,
            "height": 60,
            "justifyContent": "center",
            "marginBottom": 10.714285714285714,
            "marginLeft": 10.714285714285714,
            "marginRight": 10.714285714285714,
            "marginTop": 10.714285714285714,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Poppins-Regular",
                "fontSize": 36,
                "fontWeight": "400",
              },
            ]
          }
        >
          2
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2B303A",
            "borderRadius": 30,
            "height": 60,
            "justifyContent": "center",
            "marginBottom": 10.714285714285714,
            "marginLeft": 10.714285714285714,
            "marginRight": 10.714285714285714,
            "marginTop": 10.714285714285714,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Poppins-Regular",
                "fontSize": 36,
                "fontWeight": "400",
              },
            ]
          }
        >
          3
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "flexWrap": "wrap",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2B303A",
            "borderRadius": 30,
            "height": 60,
            "justifyContent": "center",
            "marginBottom": 10.714285714285714,
            "marginLeft": 10.714285714285714,
            "marginRight": 10.714285714285714,
            "marginTop": 10.714285714285714,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Poppins-Regular",
                "fontSize": 36,
                "fontWeight": "400",
              },
            ]
          }
        >
          4
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2B303A",
            "borderRadius": 30,
            "height": 60,
            "justifyContent": "center",
            "marginBottom": 10.714285714285714,
            "marginLeft": 10.714285714285714,
            "marginRight": 10.714285714285714,
            "marginTop": 10.714285714285714,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Poppins-Regular",
                "fontSize": 36,
                "fontWeight": "400",
              },
            ]
          }
        >
          5
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2B303A",
            "borderRadius": 30,
            "height": 60,
            "justifyContent": "center",
            "marginBottom": 10.714285714285714,
            "marginLeft": 10.714285714285714,
            "marginRight": 10.714285714285714,
            "marginTop": 10.714285714285714,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Poppins-Regular",
                "fontSize": 36,
                "fontWeight": "400",
              },
            ]
          }
        >
          6
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "flexWrap": "wrap",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2B303A",
            "borderRadius": 30,
            "height": 60,
            "justifyContent": "center",
            "marginBottom": 10.714285714285714,
            "marginLeft": 10.714285714285714,
            "marginRight": 10.714285714285714,
            "marginTop": 10.714285714285714,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Poppins-Regular",
                "fontSize": 36,
                "fontWeight": "400",
              },
            ]
          }
        >
          7
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2B303A",
            "borderRadius": 30,
            "height": 60,
            "justifyContent": "center",
            "marginBottom": 10.714285714285714,
            "marginLeft": 10.714285714285714,
            "marginRight": 10.714285714285714,
            "marginTop": 10.714285714285714,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Poppins-Regular",
                "fontSize": 36,
                "fontWeight": "400",
              },
            ]
          }
        >
          8
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2B303A",
            "borderRadius": 30,
            "height": 60,
            "justifyContent": "center",
            "marginBottom": 10.714285714285714,
            "marginLeft": 10.714285714285714,
            "marginRight": 10.714285714285714,
            "marginTop": 10.714285714285714,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Poppins-Regular",
                "fontSize": 36,
                "fontWeight": "400",
              },
            ]
          }
        >
          9
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "flexWrap": "wrap",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2B303A",
            "borderRadius": 30,
            "height": 60,
            "justifyContent": "center",
            "marginBottom": 10.714285714285714,
            "marginLeft": 10.714285714285714,
            "marginRight": 10.714285714285714,
            "marginTop": 10.714285714285714,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Poppins-Regular",
                "fontSize": 36,
                "fontWeight": "400",
              },
            ]
          }
        >
          ,
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2B303A",
            "borderRadius": 30,
            "height": 60,
            "justifyContent": "center",
            "marginBottom": 10.714285714285714,
            "marginLeft": 10.714285714285714,
            "marginRight": 10.714285714285714,
            "marginTop": 10.714285714285714,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Poppins-Regular",
                "fontSize": 36,
                "fontWeight": "400",
              },
            ]
          }
        >
          0
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 30,
              "height": 60,
              "justifyContent": "center",
              "marginBottom": 10.714285714285714,
              "marginLeft": 10.714285714285714,
              "marginRight": 10.714285714285714,
              "marginTop": 10.714285714285714,
              "width": 60,
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Icon
            color="#ffffff"
            name="backspace"
            size={24}
            style={
              Array [
                Object {
                  "marginLeft": 21.428571428571427,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 0,
            "marginTop": 64.28571428571428,
          },
        ]
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#37AF78",
            "borderRadius": 42.857142857142854,
            "borderStyle": "solid",
            "borderWidth": 2,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 10.714285714285714,
            "marginLeft": 10.714285714285714,
            "marginRight": 10.714285714285714,
            "marginTop": 10.714285714285714,
            "minWidth": 120,
            "opacity": 1,
            "paddingBottom": 21.428571428571427,
            "paddingLeft": 42.857142857142854,
            "paddingRight": 42.857142857142854,
            "paddingTop": 21.428571428571427,
          }
        }
      >
        <Text
          inverse={true}
          style={
            Array [
              Object {
                "color": "#37AF78",
                "fontFamily": "Poppins-Regular",
                "fontSize": 36,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        >
          cancel
        </Text>
      </View>
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#37AF78",
            "borderColor": "transparent",
            "borderRadius": 42.857142857142854,
            "borderStyle": "solid",
            "borderWidth": 2,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 10.714285714285714,
            "marginLeft": 10.714285714285714,
            "marginRight": 10.714285714285714,
            "marginTop": 10.714285714285714,
            "minWidth": 120,
            "opacity": 1,
            "paddingBottom": 21.428571428571427,
            "paddingLeft": 42.857142857142854,
            "paddingRight": 42.857142857142854,
            "paddingTop": 21.428571428571427,
          }
        }
      >
        <Text
          inverse={false}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Poppins-Regular",
                "fontSize": 36,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        >
          done
        </Text>
      </View>
    </View>
  </View>,
]
`;
